platforms/core-runtime/concurrent/src/main/java/org/gradle/internal/concurrent/AbstractManagedExecutor.java

```diff
@@ -25,6 +25,7 @@
 import java.util.concurrent.TimeUnit;
 
 public abstract class AbstractManagedExecutor<S extends ExecutorService> extends AbstractDelegatingExecutorService<S> implements ManagedExecutor {
+    @SuppressWarnings("ThreadLocalUsage")
     private final ThreadLocal<Object> executing = new ThreadLocal<Object>();
```

Comment:
💭 This is now needed because we've moved the class out of `:base-services` where thread local checks were disabled in ErrorProne.